cli/compiler.rs

```diff
@@ -1,26 +1,43 @@
 // Copyright 2018-2019 the Deno authors. All rights reserved. MIT license.
+use core::ops::Deref;
 use crate::isolate_state::*;
+use crate::js_errors::JSErrorColor;
 use crate::msg;
 use crate::ops;
 use crate::resources;
-use crate::resources::Resource;
 use crate::resources::ResourceId;
 use crate::startup_data;
 use crate::workers;
 use crate::workers::WorkerBehavior;
+use crate::workers::WorkerInit;
 use deno_core::deno_buf;
 use deno_core::Behavior;
 use deno_core::Buf;
+use deno_core::JSError;
 use deno_core::Op;
 use deno_core::StartupData;
+use futures::future::*;
+use futures::sync::oneshot;
 use futures::Future;
 use serde_json;
 use std::str;
 use std::sync::Arc;
 use std::sync::Mutex;
+use tokio::runtime::Runtime;
+
+pub type CompilerResult = Result<ModuleMetaData, JSError>;
+type CompilerInnerResult = Result<ModuleMetaData, Option<JSError>>;
+type WorkerErrReceiver = oneshot::Receiver<CompilerInnerResult>;
+
+#[derive(Clone)]
+struct CompilerShared {
```

Comment:
What is this shared amongst? Can you add Rust docs explaining

Reply:
Ok will do along with the logic at the end of `compile_sync`.